pkg/controller.v1beta1/experiment/manifest/generator_test.go

```diff
@@ -244,19 +217,22 @@ spec:
             - "--momentum=0.9"`
 
 	expectedRunSpec, err := util.ConvertStringToUnstructured(expectedStr)
-	if err != nil {
-		t.Errorf("ConvertStringToUnstructured failed: %v", err)
+	if diff := cmp.Diff(nil, err, cmpopts.EquateErrors()); len(diff) != 0 {
+		t.Errorf("ConvertStringToUnstructured failed (-want,+got):\n%s", diff)
 	}
 
-	tcs := []struct {
-		instance             *experimentsv1beta1.Experiment
-		parameterAssignments []commonapiv1beta1.ParameterAssignment
-		err                  bool
-		testDescription      string
+	cases := map[string]struct {
+		prepareConfigMap               func()
```

Comment:
```suggestion
		mockConfigMapGetter               *gomock.Call
```

Could you use a more strict type?

Reply:
Done.

Reply:
Done.